File: src/Application.Operation/Features/Admin/Offices/DTOs/OfficeDto.cs
using Tawtheef.Application.Common.Models;

namespace Application.Operation.Features.Admin.Offices.DTOs;

public sealed record OfficeDto
{
    public Guid Id { get; init; }
    public string NameAr { get; init; } = string.Empty;
    public string NameEn { get; init; } = string.Empty;
    public required DropdownOptions Country {get; set;
    }
    public List<DropdownOptions> SupportedCountries { get; init; } = [];
    public string AdminNameAr { get; init; } = string.Empty;
    public string AdminNameEn { get; init; } = string.Empty;
    public string AdminEmail { get; set; } = string.Empty;
    public string? PhoneCountryCode { get; init; }
    public string? PhoneNumber { get; init; }
}

File: src/Tawtheef.Websites/operations-web/src/app/pages/user/employee/offices-management/models/office.dto.ts
import {dropdownOptionsModel} from '../../../../../shared/models/dropdown-options.model';

export interface OfficeDto {
  id: string;
  nameAr: string;
  nameEn: string;
  country: dropdownOptionsModel;
  supportedCountries: dropdownOptionsModel[];
  isActive: boolean;
  adminNameAr?: string;
  adminNameEn?: string;
  adminEmail: string;
  phoneCountryCode?: string;
  phoneNumber?: string;
}
